### PR TITLE
Milestone 4: Basic UI

### PR DESCRIPTION
**Pull request by yetemegn-telaye**
### School-library-system milestone 4 Basic-UI

**In this milestone I :** 

- [ ] Created app class to hold the state of my app and run logic
- [ ] Create main class as an entry point 
- [ ] Run the program from console

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+  def create_person
+    puts 'What do you want to create, a Teacher (T) or a Student (S)? [Enter the Letter]:'
+    input_type = gets.chomp.upcase
+    case input_type
+    when 'T'
+      create_teacher
+    when 'S'
+      create_student
+    else
```
Comment (AmaduKamara) [inline app.rb:39]: - [x] When I try to create a teacher or student, the app crashes with the errors shown in the screenshot below. Kindly take a look at the arguments passed and fix the issue so a person can be created in order for the rental feature to work 👍🏽

Screenshot 2023-03-17 at 10 09 30 PM

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    puts 'Select a person that wants to rent a book (choose a number)'
+    people.each_with_index do |index, person|
+      puts "#{index} [#{person.class}] Name: #{person.name}, Age: #{person.age}, ID: #{person.id}"
+    end
+    person_index = gets.chomp.to_i
+    puts 'Enter Date: '
+    date = gets.chomp.to_s
+    rentals << Rental.new(date, people[person_index], books[book_index])
+    puts 'Book successfully rented'
+  end
```
Comment (AmaduKamara) [inline app.rb]: - [x] As I was testing the other features, a book can be created but when I try creating a rental, the app crashes. Please fix the issue on that so a rental is created 👍🏽

Screenshot 2023-03-17 at 10 04 45 PM

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @yetemegn-telaye,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ You used Gitflow correctly 👍🏽
✅ Linter checks passed 👍🏽
✅ The app UI looks nice and interactive 👍🏽

## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly work on the features for creating a student and a teacher, so a rental of a book can be created for them successfully 👍🏽

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app.rb (hunk 1 above)
Comment (Whoistolu) [reply app.rb:39]: - Well resolved @yetemegn-telaye 👍🏽

File: app.rb (hunk 2 above)
Comment (Whoistolu) [reply app.rb]: - [ ] [REQUIRED] Hello @yetemegn-telaye ,

I encountered a `noMethodError` while trying to create a rental. The `name` method could not be identified in the `create_rental` method. Let's fix that 👍🏽 

https://github.com/yetemegn-telaye/school-library-system/blob/38164bc9c788a05996c7523c032b420ca62fc7ac/app.rb#L95

Screenshot 2023-03-18 at 15 15 19

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @yetemegn-telaye 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 2 above)
Comment (Whoistolu) [reply app.rb]: Hi @yetemegn-telaye 
- Great job resolving this 👍🏽 
Screenshot 2023-03-18 at 15 40 45

**Review by Whoistolu — APPROVED** [review]
### Hi @yetemegn-telaye 

Great job fixing the previous recommendations 👍🏽 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**